Scenario.cs

```diff
@@ -124,13 +125,23 @@ private CosmosClient CreateCosmosClient(bool bulkExecution)
                 MaxRetryWaitTimeOnRateLimitedRequests = TimeSpan.FromSeconds(60)
             };
 
-            CosmosClient cosmosClient = new CosmosClient(
-                accountEndpoint: this.Configurations["AppSettings:accountEndpoint"],
-                authKeyOrResourceToken: this.Configurations["AppSettings:authKeyOrResourceTokenCredential"],
-                clientOptions: cosmosClientOptions
+            bool useAADAuth = Convert.ToBoolean(this.Configurations["AppSettings:useAADAuth"]);
+            if (useAADAuth) 
+            {
+                return new CosmosClient(
+                    accountEndpoint: this.Configurations["AppSettings:accountEndpoint"],
+                    tokenCredential: new DefaultAzureCredential(),
+                    clientOptions: cosmosClientOptions

```

Comment:
how will token credential for AAD auth be acquired?

Reply:
This is handled by '_DefaultAzureCredential_'. Please see : https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet